
Ключевое слово **throw** используется для явного выброса исключения в коде. Это способ сигнализировать о том, что в программе произошла ошибка или возникла исключительная ситуация, требующая обработки.

- Когда выполнение кода доходит до оператора **throw**, текущий метод немедленно прерывается, а управление передаётся в ближайший блок [[Try catch|catch]], способный обработать выброшенное исключение.
- С помощью **throw** можно выбросить как стандартные исключения (*например, IllegalArgumentException*), так и [[Создание исключений|пользовательские]].

**Пример:**
![[Pasted image 20250322223724.png]]
**Вывод:**
![[Pasted image 20250322223906.png]]
- В методе **checkAge** проверяется условие: если возраст меньше 18, выбрасывается исключение **IllegalArgumentException** с сообщением.
- В методе **main** это исключение перехватывается в блоке **try-catch**, и выводится сообщение об ошибке.


